sound/soc/codecs/rt700.c

```diff
@@ -1339,7 +1326,33 @@ static int rt700_pcm_hw_params(struct snd_pcm_substream *substream,
 	snd_soc_component_write(component, RT700_ADC_FORMAT_L, val);
 
 	dev_err(dai->dev, "5 %s %s", __func__, dai->name);
-	return retval;
+	return 0;
+}
+
+static int rt700_prepare_stream(struct snd_pcm_substream *substream,
+				struct snd_soc_dai *dai)
+{
+	struct snd_soc_component *component = dai->component;
+	struct rt700_priv *rt700 = snd_soc_component_get_drvdata(component);
+	struct sdw_stream_data *stream;
+	int retval;
+
+	stream = snd_soc_dai_get_dma_data(dai, substream);
+
+	if (!stream) {
+		dev_err(dai->dev, "no dma data found in dai %s", dai->name);
+		return -ENOMEM;
+	}
+
+	retval = sdw_stream_add_slave(rt700->slave, &rt700->stream_config,
+				      &rt700->port_config, 1,
+				      stream->sdw_stream);
+	if (retval) {
+		dev_err(dai->dev, "Unable to configure port\n");
+		return retval;
```

Comment:
@RanderWang just return retval outside the if. No need for 2 separate return statements here